t/re-gmatch.t

```diff
@@ -0,0 +1,509 @@
+            ngx.say("$1: ", m2[1])
+            ngx.say("$2: ", m2[2])
+            ngx.say("$3: ", m2[3])
+        }
+    }
+--- request
+GET /re
+--- response_body
+matched: hello
+$1: h
+$2: ello
+$3: nil
+matched: world
+$1: w
+$2: orld
+$3: nil
+--- error_log eval
+qr/\[TRACE\s+\d+\s+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: named submatches
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local m1, m2
+            local gmatch = ngx.re.gmatch
+            for _ = 1, 100 do
+                local iter = gmatch("hello,world", [[(?<first>\w)(\w+)]], "jo")
+                m1 = iter()
+                m2 = iter()
+            end
+            ngx.say("matched: ", m1[0])
+            ngx.say("$1: ", m1[1])
+            ngx.say("$2: ", m1[2])
+            ngx.say("$first: ", m1.first)
+            ngx.say("$second: ", m1.second)
+            ngx.say("matched: ", m2[0])
+            ngx.say("$1: ", m2[1])
+            ngx.say("$2: ", m2[2])
+            ngx.say("$first: ", m2.first)
+            ngx.say("$second: ", m2.second)
+        }
+    }
+--- request
+GET /re
+--- response_body
+matched: hello
+$1: h
+$2: ello
+$first: h
+$second: nil
+matched: world
+$1: w
+$2: orld
+$first: w
+$second: nil
+--- error_log eval
+qr/\[TRACE\s+\d+\s+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: unmatched captures are false
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local iter = ngx.re.gmatch(
+                "hello! world!", [[(\w+)(, .+)?(!)]], "jo")
+            if iter then
+                while true do
+                    local m = iter()
+                    if not m then
+                        return
+                    end
+                    ngx.say(m[0])
+                    ngx.say(m[1])
+                    ngx.say(m[2])
+                    ngx.say(m[3])
+                end
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+hello!
+hello
+false
+!
+world!
+world
+false
+!
+--- error_log eval
+qr/\[TRACE\s+\d+\s+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 9: unmatched trailing captures are false
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local iter = ngx.re.gmatch("hello", [[(\w+)(, .+)?(!)?]], "jo")
+            if iter then
+                while true do
+                    local m = iter()
+                    if not m then
+                        return
+                    end
+                    ngx.say(m[0])
+                    ngx.say(m[1])
+                    ngx.say(m[2])
+                    ngx.say(m[3])
+                end
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+hello
+hello
+false
+false
+--- error_log eval
+qr/\[TRACE\s+\d+\s+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 10: unmatched name captures are false
```

Comment:
minor typo: I think you meant `named captures`?

Reply:
Good catch +1